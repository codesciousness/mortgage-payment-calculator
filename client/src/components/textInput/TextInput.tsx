import React from 'react';
import TextField from '@mui/material/TextField';

type TextInputProps = {
    name: string;
    value: string;
    width?: number | string;
    placeholder?: string;
    error?: string;
    onChange: () => void;
};

const TextInput = ({ name, value, placeholder, width, error, onChange }: TextInputProps): JSX.Element => {
    const id = name.replaceAll(' ', '');

    return (
        <TextField
            id={id}
            name={name}
            label={name}
            placeholder={placeholder ? placeholder : `Enter your ${name.toLowerCase()}`}
            onChange={onChange}
            helperText={error}
            size='small'
            sx={{
                width,
                margin: '0.25rem 0',
                '& .MuiOutlinedInput-root': {
                    '&.Mui-focused fieldset': {
                        borderColor: 'DeepSkyBlue',
                    }
                }
            }}
        >
            {value}
        </TextField>
    );
};

export default TextInput;