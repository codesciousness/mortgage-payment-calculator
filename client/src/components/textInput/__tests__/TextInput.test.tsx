import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import TextInput from '../TextInput';

const name = 'Label Name';
const value = 'Test Value';
const width = 300;
const placeholder = 'Placeholder Text';
const error = 'Error!';
const onChange = () => null;

describe('<TextInput />', () => {
    test('TextInput component is defined', () => {
        expect(<TextInput name={name} value={value} onChange={onChange}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<TextInput name={name} value={value} width={width} 
            placeholder={placeholder} onChange={onChange}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('renders correctly with props', () => {
        render(<TextInput name={name} value={value} width={width} 
            placeholder={placeholder} onChange={onChange}/>);
        expect(screen.getByLabelText('Label Name')).toBeInTheDocument();
        expect(screen.getByDisplayValue('Test Value')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('Placeholder Text')).toBeInTheDocument();
    });
    test('renders correctly when error prop is provided', () => {
        render(<TextInput name={name} value={value} width={width} 
            placeholder={placeholder} error={error} onChange={onChange}/>);
        expect(screen.getByText('Error!')).toBeInTheDocument();
    });
});