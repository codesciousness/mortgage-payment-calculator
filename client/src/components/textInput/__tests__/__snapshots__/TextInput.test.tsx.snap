// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> component matches the snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-qhedn9-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    htmlFor="LabelName"
    id="LabelName-label"
  >
    Label Name
  </label>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id="LabelName"
      name="Label Name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Placeholder Text"
      required={false}
      type="text"
      value="Test Value"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1in441m"
      >
        <span>
          Label Name
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
